tools/ios/build_xcframework.py

```diff
@@ -179,6 +191,20 @@ def parse_args():
         required=True,
         help="Path to output directory.",
     )
+    # Same as how android works
+    # This one is used in the ci pipeline for accelerating the build process,
+    # We have 4 archs to be build. It's in sequence by default, but we can build them in parallel.
+    # The parallel build works as:
+    #   1. build the lib files for each arch in different ci jobs
+    #   2. download all the lib files in tasks
+    #   2. pack the xcframework package in ci task
```

Comment:
it's a good idea to allow parallelization of the framework builds. however, this needs some refinement for ios.

there is a framework (for one platform, for one arch), and then a fat framework (for one platform, for all archs), and then an xcframework (for all platform/archs).

so for example, you'd need to build each framework for one platform (e.g., iphonesimulator/x86_64 and iphonesimulator/arm64), then put all those together in one fat framework with `lipo` for the iphonesimulator platform. then, put the fat frameworks for each platform (iphonesimulator and iphoneos) together in one xcframework with `xcrun xcodebuild -create-xcframework ...`.

Reply:
FYI working on this in a branch: https://github.com/microsoft/onnxruntime-extensions/tree/edgchen1/split_xcframework_build